akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/ResponseParsing.scala

```diff
@@ -8,15 +8,32 @@ package client
 import akka.annotation.InternalApi
 import akka.http.impl.engine.http2.RequestParsing._
 import akka.http.impl.engine.parsing.HttpHeaderParser
+import akka.http.impl.engine.server.HttpAttributes
+import akka.http.scaladsl.model
 import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.`Tls-Session-Info`
+import akka.http.scaladsl.settings.ParserSettings
+import akka.stream.Attributes
 import akka.util.OptionVal
 
 import scala.annotation.tailrec
 import scala.collection.immutable.VectorBuilder
 
 @InternalApi
 private[http2] object ResponseParsing {
-  def parseResponse(httpHeaderParser: HttpHeaderParser): Http2SubStream => HttpResponse = { subStream =>
+  def parseResponse(httpHeaderParser: HttpHeaderParser, settings: ParserSettings, attributes: Attributes): Http2SubStream => HttpResponse = { subStream =>
+
+    val tlsSessionInfoHeader: Option[`Tls-Session-Info`] =
```

Comment:
I considered not doing that but because synthetic headers being deprecated, but there is no deprecation notice on the `tls-session-info-header` so I thought it was important to keep it for some reason.

I don't have a strong opinion, as long as the attribute one works on both (and we document that the header one is deprecated/does not work with HTTP/2)

Reply:
Yeah, that's right. Should we remove the header feature at some point, it's easy enough to remove it here.